### PR TITLE
[Drupal] Treat Drupal pages as production-ready pages

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
- Fails the build if the Metalsmith-Drupal build step fails
- Enables the broken link checker on Drupal pages

Ticket - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17451
Ticket for removing `/drupal` from Staging - https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17454

## Acceptance criteria
- [ ] Drupal pages are treated as production-ready

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/site/stages/build/drupal/metalsmith-drupal.js [hunk 1]
```diff
@@ -142,9 +147,11 @@ function getDrupalContent(buildOptions) {
       log(err.stack);
       log(JSON.stringify(drupalData));
       log('Failed to pipe Drupal content into Metalsmith!');
-      log('Continuing with build anyway...');
-      // done(err);
-      done();
+      if (buildOptions.buildtype !== ENVIRONMENTS.LOCALHOST) {
+        done(err);
+      } else {
+        done();
```
Comment (jbalboni) [inline src/site/stages/build/drupal/metalsmith-drupal.js]: If we don't do this, then devs who don't care about Drupal content will have to be on the SOCKS proxy. Also, Heroku will stop working, I think.

File: src/site/constants/drupals.js [hunk 2]
```diff
@@ -37,8 +37,9 @@ const ENABLED_ENVIRONMENTS = new Set([
 ]);
 
 const PREFIXED_ENVIRONMENTS = new Set([
-  ENVIRONMENTS.LOCALHOST,
-  ENVIRONMENTS.VAGOVSTAGING,
+  // ENVIRONMENTS.LOCALHOST,
```
Comment (jbalboni) [inline src/site/constants/drupals.js]: I assume we can remove this for now?

Comment (ncksllvn) [reply src/site/constants/drupals.js]: The only issue with localhost is that Drupal Staging provides all of the hub landing pages, and content is still editing some them. I think this is still the right thing to do, but that's something we'll have to deal with.

Added this ticket to help with handling Heroku, https://github.com/department-of-veterans-affairs/vets.gov-team/issues/17537.

File: src/site/stages/build/drupal/graphql/GetAllPages.graphql.js [hunk 3]
```diff
@@ -26,7 +26,11 @@ module.exports = `
   ${bioPage}
 
   query GetAllPages($today: String!) {
-    nodeQuery(limit: 500) {
+    nodeQuery(limit: 500, filter: {
+      conditions: [
+        { field: "status", value: ["1"] }
+      ]
+    }) {
```
Comment (ncksllvn) [inline src/site/stages/build/drupal/graphql/GetAllPages.graphql.js]: 🔥

**Review by jbalboni — APPROVED** [review]

**Pull request merged**